### PR TITLE
728 rmc 319b generate ce spg initial contact letter unaddressed print files ew

### PR DESCRIPTION
**Pull request by ryangrundy7**
# Motivation and Context

For Census 2021 we're going to have to create unaddressed CE/SPG initial contact letters for QM and PPO. To be able to do this, qid-Batch-runner had to be updated to be able to cope with sending batches to different locations. This has been done by splitting the batches in suppliers and types and updating the code so when we pass in the PPO supplier, it'll put the batch into the correct SFTP directory.

# What has changed

- Now able to pass in a supplier when generating print files to decide which sftp location the files will go to.
- Added mappings for suppliers to keys and sftp directories.
- Added updated tests
# How to test?

- Run `make test` and all should pass.
- Generate a PPO print files and pass in the PPO supplier. It should generate files into the ppo_dev sftp directory.

# Links

[Trello](https://trello.com/c/Ij81s0mo/)

### CODE REVIEW

**Comment by ryangrundy7** [comment]
[RMC-319B Generate CE/SPG Initial contact letter Unaddressed Print Files (E+W) (8)](https://trello.com/c/Ij81s0mo/728-rmc-319b-generate-ce-spg-initial-contact-letter-unaddressed-print-files-ew-8)

File: qid-batch-runner.yml [hunk 1]
```diff
@@ -101,8 +106,10 @@ spec:
             value: "/home/qidbatchrunner/.sftp-ssh/id_rsa"
           - name: OUR_PUBLIC_KEY_PATH
             value: "/home/qidbatchrunner/.pgp-keys/our_public_key.asc"
-          - name: OTHER_PUBLIC_KEY_PATH
-            value: "/home/qidbatchrunner/.pgp-keys/other_public_key.asc"
+          - name: QM_PUBLIC_KEY_PATH
+            value: "/home/qidbatchrunner/.pgp-keys/supplier_QM_dummy_public.asc"
+          - name: PPO_PUBLIC_KEY_PATH
+            value: "/home/qidbatchrunner/.pgp-keys/supplier_ppo_dummy_public.asc"
```
Comment (NickGrantX) [inline qid-batch-runner.yml]: would be nice if these keys were consistently named

File: mappings.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+SUPPLIER_TO_KEY_PATH = {
+    'QM': os.getenv('QM_PUBLIC_KEY_PATH') or str(
+        Path(__file__).parents[0].joinpath('dummy_keys', 'supplier_QM_dummy_public.asc')),
+    'PPO': os.getenv('PPO_PUBLIC_KEY_PATH') or str(
+        Path(__file__).parents[0].joinpath('dummy_keys', 'dummy_ppo_supplier_public_key.asc'))
+}
+
+SUPPLIER_TO_PRIVATE_KEY_PATH = {
+    'QM': 'dummy_keys/supplier_QM_dummy_private.asc',
```
Comment (NickGrantX) [inline mappings.py]: can we make these key names consistent, please

File: mappings.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    'PPO': 'dummy_keys/dummy_ppo_supplier_private_key.asc'
```
Comment (NickGrantX) [inline mappings.py]: `dummy_ppo_supplier_private_key.asc` should be renamed to `supplier_PPO_dummy_private.asc`

File: qid-batch-runner.yml (hunk 1 above)
Comment (NickGrantX) [reply qid-batch-runner.yml]: `dummy_ppo_supplier_public_key.asc` should be renamed to `supplier_PPO_dummy_public.asc`

File: qid-batch-runner.yml [hunk 4]
```diff
@@ -140,4 +147,6 @@ spec:
               - key: our-public-key
                 path: "our_public_key.asc"
               - key: qm-supplier-public-key
-                path: "other_public_key.asc"
+                path: "supplier_QM_dummy_public.asc"
+              - key: ppo-supplier-public-key
+                path: "supplier_ppo_dummy_public.asc"
```
Comment (NickGrantX) [inline qid-batch-runner.yml]: `dummy_ppo_supplier_public_key.asc` should be renamed to `supplier_PPO_dummy_public.asc`

File: acceptance_tests/steps/valid_print_files.py [hunk 5]
```diff
@@ -61,30 +62,33 @@ def rabbit_connection_and_channel():
...
+        context.batch_config_path, Path('.'), context.batch_id, supplier)
 
 
-@then('the "{manifest_count}" print files contents are valid')
-def validate_print_file_data(context, manifest_count):
+@then('the "{manifest_count}" print files contents are valid for the "{supplier}"')
+def validate_print_file_data(context, manifest_count, supplier):
     manifests = [file_path for file_path in context.print_file_paths if file_path.suffix == '.manifest']
-    print_files = [file_path for file_path in context.print_file_paths if file_path.suffix == '.csv.gpg']
+    print_files = [file_path for file_path in context.print_file_paths if file_path.suffix == '.gpg']
```
Comment (AdamHawtin) [inline acceptance_tests/steps/valid_print_files.py]: Why this change? I don't see why it shouldn't check the `.csv` part of the suffix, it's part of the expected name format.

Comment (ryangrundy7) [reply acceptance_tests/steps/valid_print_files.py]: I think this bit has always been broke. If you run the test and put a breakpoint there, the list comes back as empty but still passes. I'll look at checking both parts of it though because it really should

Comment (AdamHawtin) [reply acceptance_tests/steps/valid_print_files.py]: Perhaps the `.suffix` field doesn't pick it up, `.name.endswith` might be more suitable?

Comment (ryangrundy7) [reply acceptance_tests/steps/valid_print_files.py]: Could possibly change it to this 
```python
print_files = [file_path for file_path in context.print_file_paths if '.csv' and '.gpg' in file_path.suffixes]
```

Comment (AdamHawtin) [reply acceptance_tests/steps/valid_print_files.py]: Worth asserting the length is expected too, since it was slipping through empty before

Comment (AdamHawtin) [reply acceptance_tests/steps/valid_print_files.py]: Actually that logic needs a bit of a tweek, it's doing 
```python
if ('.csv') and ('.gpg' in suffixes)
```
rather than checking both strings are in `suffixes`

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

**Review by hbrace — APPROVED** [review]

**Review by AdamHawtin — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -87,7 +87,9 @@ Then run
 pipenv run python generate_print_files.py <path to config csv> <path to output directory> <supplier> <batch ID> --no-gcs
 ```
 
-This will read the generated qid/uac pairs and generate the print and manifest files in the specified supplier directory
+This will read the generated qid/uac pairs and generate the print and manifest files in the specified supplier directory. 
+Make sure to pass in the correct supplier so the files go to the correct location. The suppliers have been put onto the
```
Comment (gavinleeedwards) [inline README.md:91]: Might be handy to specify what the options are for supplier are, given that there are only 2 allowed options and it's case sensitive.  No biggie though.
